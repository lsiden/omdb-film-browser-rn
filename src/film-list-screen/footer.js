import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import { View, StyleSheet, ActivityIndicator } from "react-native"

import { primaryColor } from "constants"

const style = StyleSheet.create({
  wrapper: {
    alignItems: "center",
    paddingVertical: 20,
    borderTopWidth: 1
  }
})

export const footer = ({ isFetching }) =>
  isFetching && (
    <View style={style.wrapper}>
      <ActivityIndicator size="large" color={primaryColor} />
    </View>
  )
footer.propTypes = {
  isFetching: PropTypes.bool
}
footer.defaultProps = {
  isFetching: false
}

export default connect(state => ({
  isFetching: state.isFetching
}))(footer)
