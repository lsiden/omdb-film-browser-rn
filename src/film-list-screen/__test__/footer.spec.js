import React from "react"
import renderer from "react-native-test-utils"

import { footer as Footer } from "../footer"

const createWrapper = (props = {}) => renderer(<Footer {...props} />)

it("renders nothing if isFetching is false", () => {
  const wrapper = createWrapper({ isFetching: false })
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it("renders ActivityIndicator if isFetching is true", () => {
  const wrapper = createWrapper({ isFetching: true })
  expect(wrapper.toJSON()).toMatchSnapshot()
})
