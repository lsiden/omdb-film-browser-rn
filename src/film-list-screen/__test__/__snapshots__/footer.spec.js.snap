// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ActivityIndicator if isFetching is true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopWidth": 1,
      "paddingVertical": 20,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#05347f"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`renders nothing if isFetching is false 1`] = `null`;
